packages/expansion-hub/src/internal/ExpansionHub.ts

```diff
@@ -8,26 +8,42 @@ import {
     Serial, VerbosityLevel, Version
 } from "@rev-robotics/rhsplib"
 import {RevHubType} from "../RevHubType";
+import {ParentRevHub, RevHub} from "../RevHub";
 import {EventEmitter} from "events";
-import {RevHub} from "../RevHub";
 
 export class ExpansionHubInternal implements ExpansionHub {
-    constructor() {
+    constructor(isParent: boolean, serialNumber: string | undefined = undefined) {
```

Comment:
```suggestion
    constructor(isParent: boolean, serialNumber?: string) {
```

Reply:
In fact, we can go one step further and enforce that a serial number can only be specified on a parent:

```suggestion
    constructor(isParent: true, serialNumber: string);
    constructor(isParent: false);
    constructor(isParent: boolean, serialNumber?:) {
```

The way overloads work in Typescript is that you can specify multiple signatures for a function, but only one implementation, and the implementation has to have a signature that is compatible with all of the signature definitions. However, the compiler will only permit parameters that match one of the explicit signature definitions.